### PR TITLE
Self-registering server recipe

### PR DESCRIPTION
**Pull request by wolf31o2**
Registers a server as a SSH datasource using a given SDM admin key which has
rights to perform the necessary actions.

Signed-off-by: Chris Gianelloni <cgianelloni@applause.com>

### CODE REVIEW

**Comment by wolf31o2** [comment]
Relay: 
image

Server:
image

File: recipes/server.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  cwd Chef::Config['file_cache_path']
+  environment('SDM_ADMIN_TOKEN' => node['strongdm']['admin_token'])
+  creates '/root/.sdm/auth.json'
+end
+
+execute 'sdm-admin-add-servers' do
+  command "./sdm admin servers add -p #{node['fqdn']} #{node['strongdm']['user']}@#{node['ipaddress']}"
+  cwd Chef::Config['file_cache_path']
+  environment('SDM_ADMIN_TOKEN' => node['strongdm']['admin_token'])
+  not_if "#{Chef::Config['file_cache_path']}/sdm admin servers list | grep #{node['fqdn']}"
```
Comment (agaffney) [inline recipes/server.rb]: Why use the file cache path explicitly here when calling `sdm`? Does the `cwd` param not apply to the `not_if` command? Is this a "better safe than sorry" thing?

File: recipes/server.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+directory '/opt/strongdm/.ssh' do
+  recursive true
+  owner node['strongdm']['user']
+  group node['strongdm']['user']
+  mode 0700
+  notifies :run, 'execute[sdm-ssh-pubkey]', :immediately
+end
+
+execute 'sdm-ssh-pubkey' do
+  command "./sdm ssh pubkey #{node['fqdn']} | tee -a /opt/strongdm/.ssh/authorized_keys"
```
Comment (agaffney) [inline recipes/server.rb]: Why only have this run when the above resource changes something (which won't be often)? Will this value not change?

File: recipes/server.rb (hunk 1 above)
Comment (wolf31o2) [reply recipes/server.rb]: Better safe than sorry. I couldn't find the answer in the docs and didn't feel like trial by fire.

File: recipes/server.rb (hunk 2 above)
Comment (wolf31o2) [reply recipes/server.rb]: Correct, it will not change. I needed some way to ensure this doesn't append every Chef run.

**Pull request merged**